src/runtime/hexagon_remote/host_malloc.cpp

```diff
@@ -173,22 +293,39 @@ void *halide_hexagon_host_malloc(size_t size) {
 
     int buf_fd = 0;
     ion_user_handle_t handle = 0;
-    handle = ion_alloc(ion_fd, size, alignment, 1 << heap_id, ion_flags);
-    if (handle < 0) {
-        __android_log_print(ANDROID_LOG_ERROR, "halide", "ion_alloc(%d, %lld, %lld, %d, %d) failed",
-                            ion_fd, (long long)size, (long long)alignment, 1 << heap_id, ion_flags);
-        return NULL;
-    }
-    // Map the ion handle to a file buffer.
-    if (use_newer_ioctl) {
-        buf_fd = handle;
-    } else {
-        buf_fd = ion_map(ion_fd, handle);
+
+    if (use_libdmabuf) {
+        buf_fd = dmabuf_alloc_fn(dmabufAllocator, dmabuf_heap, size, 0, 0);
         if (buf_fd < 0) {
-            __android_log_print(ANDROID_LOG_ERROR, "halide", "ion_map(%d, %d) failed", ion_fd, handle);
-            ion_free(ion_fd, handle);
+            __android_log_print(ANDROID_LOG_ERROR, "halide", "DmabufHeapAlloc(%p, \"%s\", %zd, %d, %d) failed",
+                                dmabufAllocator, dmabuf_heap, size, 0, 0);
+            return NULL;
+        }
+    } else if (use_libion) {
+        buf_fd = ion_alloc(ion_fd, size, alignment, 1 << heap_id, ion_flags);
+        if (buf_fd < 0) {
+            __android_log_print(ANDROID_LOG_ERROR, "halide", "ion_alloc(%d, %zd, %zd, %d, %d) failed",
+                                ion_fd, size, alignment, 1 << heap_id, ion_flags);
+            return NULL;
+        }
+    } else {  // !use_libion
+        handle = ion_alloc(ion_fd, size, alignment, 1 << heap_id, ion_flags);
```

Comment:
This looks identical to the `if (use_libion)` case?

Reply:
The difference (for !use_libion) is further below in lines 318-328